types/example.pp

```diff
@@ -1,14 +1,14 @@
 type Patterndb::Example = Struct[
   {
-    program      => String[1],
+    program      => String,
```

Comment:
/me reads xsd :exploding_head: 

If I am correct, the schema says that *the attribute is optional* but says nothing about *its length*.  I tried to remove the program from one of my examples, and validation pass but the test ignore the example I modified:

```
NOT Testing message as program is unset: ...
```

Of course, setting program to an empty string fail validation, which I expected.


If we want to allow empty string, I am in favor of being explicit about it: `String[0]` (After some years of work in numerous modules, I came to the conclusion that I should consider all usage of `String` without length as a code smell :smile:)

If we want to stick to the XML schema, we can also make this optional (will write about it bellow), but as far as I am concerned, it feels broken.

Reply:
FTR leaving the program out is equivalent to a wildcard for all programs : the rule will match all programs